src/dra/7A4D0.c

```diff
@@ -721,7 +721,641 @@ bool func_8011BD48(Entity* entity) {
 }
 
 // player turns white for some sort of status effect
-INCLUDE_ASM("dra/nonmatchings/7A4D0", EntityPlayerBlinkWhite);
+void EntityPlayerBlinkWhite(Entity* self) {
+    Primitive* prim;
+    u8 var_s7;
+    u8 var_s6;
+    u8 sp7f;
+    u8 sp7e;
+    s16 sp7c;
+    s16 sp7a;
+    s16 var_s2;
+    s16 var_s5;
+    s16 sp78;
+    s16 sp76;
+    s16 sp74;
+    s16 sp72;
+    s16 var_s3;
+    s16 var_s8;
+    s32 sp54;
+    s32 sp50;
+    s16 sp70;
+    s16* sp4c;
+    s16 sp6e;
+    s16 sp6c;
+    s16 sp6a;
+    s16 sp68;
+    s16 sp66;
+    s16 sp64;
+    s16 sp62;
+    s16 sp60;
+    s16* sp5c;
+    u8* sp58;
+    s32 sp48;
+    s32 sp44;
+    s32 sp40;
+    Entity* sp3C;
+    s32 sp38;
+    s32 var_s1;
+
+    sp70 = (self->params & 0x7F00) >> 8;
+    sp48 = 0;
+    if ((PLAYER.animSet == 0xF) && (sp70 == 0x23) && g_Player.unk66 < 2) {
+        sp48 = 1;
+    }
+    if ((((sp70 & 0x3F) != 0x1D) &&
+         (g_Player.unk0C & PLAYER_STATUS_MIST_FORM)) ||
+        (g_Player.unk0C & PLAYER_STATUS_AXEARMOR)) {
+        goto block_229;
+    }
+    if ((g_Player.unk6C) && sp70 != 0x20 && sp70 != 0x21 &&
+        ((sp70 & 0x3F) != 0x1D)) {
+        g_Player.unk6E = 0;
+        goto block_231;
+    }
+    if ((g_Player.unk6E) && sp70 != 0x23 && sp70 != 0x24 &&
+        ((sp70 & 0x3F) != 0x1D)) {
+        g_Player.unk6C = 0;
+        goto block_231;
+    }
+    if (((PLAYER.step == Player_SpellHellfire) && (PLAYER.palette == 0x810D)) ||
+        (!PLAYER.animSet) || !(PLAYER.animCurFrame & 0x7FFF)) {
+        goto block_229;
+    }
+    var_s8 = self->posY.i.hi = PLAYER.posY.i.hi;
+    var_s3 = self->posX.i.hi = PLAYER.posX.i.hi;
+    self->facingLeft = PLAYER.facingLeft;
+    if (PLAYER.animSet == 1) {
+        sp5c = D_800CF324[PLAYER.animCurFrame & 0x7FFF];
+    }
+    if (PLAYER.animSet == 0xD) {
+        sp5c = D_800CFE48[PLAYER.animCurFrame & 0x7FFF];
+    }
+    if (PLAYER.animSet == 0xF) {
+        if (sp48 != 0) {
+            if (D_801396E0 == 0xD) {
+                sp5c = D_800CFE48[D_801396EC & 0x7FFF];
+#ifdef VER_PSP
+                if (sp5c) {
+#endif
+                    sp72 = *sp5c++;
+                    sp72 &= 0x7FFF;
+                    sp58 = (*g_PlOvlAluBatSpritesheet)[sp72];
+#ifdef VER_PSP
+                } else {
+                    sp58 = 0;
+                }
+#endif
+            } else {
+                sp5c = D_800CF324[D_801396EC & 0x7FFF];
+#ifdef VER_PSP
+                if (sp5c) {
+#endif
+                    sp72 = *sp5c++;
+                    sp72 &= 0x7FFF;
+                    sp58 = ((u8**)SPRITESHEET_PTR)[sp72];
```

Comment:
Is this still necessary after the recent maspsx fix?

Reply:
My understanding was that the maspsx fix was in order to allow this to work. Previously even this would not work. Simply doing g_PlOvlSpritesheet[sp72] maintains the old issue. with $at inversion.

Reply:
I haven't been able to get it to work without using this setup, so I think we go with this, and leave it open for cleaning up later, if we want to. I can add a comment if you like.